### PR TITLE
Add money to a savings goal

### PR DESCRIPTION
**Pull request by pettomartino**
Hi Guys,

I'm not sure about the development process of this SDK, if it follows the documentation's version, API or something like that but I'm implementing a rounding to savings feature for my personal use and I needed this function, so I developed it. I followed your design decisions and the pattern you're using on the SDK, I hope that can be useful.

### CODE REVIEW

**Comment by Stephan5** [comment]
Hey pettomartino! 
Thank you for adding this, we'll have a look and try and get it in.

File: src/entities/savingsGoals.js [hunk 1]
```diff
@@ -97,6 +97,31 @@ class SavingsGoals {
...
+    typeValidation(arguments, addMoneySavingsGoalParameterDefinition);
+    const url = `${this.options.apiUrl}/api/v1/savings-goals/${savingsGoalId}/add-money/${transactionId}`;
+    log(`PUT ${url}`);
+    return axios({
+      method: 'PUT',
+      url,
+      headers: postHeaders(accessToken),
+      data: JSON.stringify({
+        amount: {
+          currency: 'GBP',
```
Comment (bytelucas) [inline src/entities/savingsGoals.js]: The currency shouldn't be hardcoded to GBP, but instead there should be a `currency` param. If not present, this should default to GBP.

File: test/entities/savingsGoals.spec.js [hunk 2]
```diff
@@ -107,4 +109,27 @@ describe('Savings Goals - ', function () {
       .catch(done);
   });
 
+
+  nock('http://localhost:8080', expectAuthorizationHeader(accessToken))
+    .put(
+      `/api/v1/savings-goals/${savingsGoalId}/add-money/${transactionId}`,
+      {
+        "amount": {
+          "currency": "GBP",
```
Comment (bytelucas) [inline test/entities/savingsGoals.spec.js]: As before, currency shouldn't be hardcoded to GBP.

File: src/starling.js [hunk 3]
```diff
@@ -220,6 +220,18 @@ class Starling {
...
 
+  /**
+   * Add money to a specific savings goal
+   * @param {string} accessToken - the oauth bearer token.
+   * @param {string} savingsGoalId - the savings goal's ID.
+   * @param {string} transactionId - a transaction ID for this transaction
+   * @param {number} amount - an amount in minor unit
+   * @return {Promise} - the http request promise
+   */
+  addMoneyToSavingsGoal (accessToken = this.config.accessToken, savingsGoalId, transactionId, amount) {
```
Comment (bytelucas) [inline src/starling.js]: This should also have a `currency = 'GBP'` parameter

**Comment by bytelucas** [comment]
Just a couple bits that need changing in regards to currency, but nice work! 😄

**Comment by pettomartino** [comment]
@bytelucas Oh gosh, what a mistake. 
Should be fixed now.

Thank you for reviewing.

**Comment by pettomartino** [comment]
@bytelucas do you have any feedback?

**Pull request merged**